### ISSUE TITLE
Support psr/simple-cache 3.0

### ISSUE THREAD

**Issue by brayniverse** [CuyZ/Valinor#25]
Hey, I was wondering whether you want to support psr/simple-cache 3.0? It would require dropping support for PHP 7.4 which you might not want to do. I'm happy to submit a PR to drop support for PHP 7.4, and then another to add the return types to your cache implementations and delete `stubs/Psr/SimpleCache/CacheInterface.stub` and the reference to it in `phpstan.neon.dist`.

**Comment by romm** [comment]
Hi @brayniverse, I noticed that when I added support for `psr/simple-cache:^2.0`. You're right, supporting v3.0 would mean dropping PHP 7.4 support.

While I plan to do it at some time, that would mean maintaining two separate branches, because the old branch would live at least until the end of PHP 7.4 security support, on [28 nov. 2022](https://www.php.net/supported-versions.php).

What is the reason that forces you to have `psr/simple-cache:^3.0` support on your side? Are there alternatives?
